src/components/KTreeList/KTreeList.vue

```diff
@@ -0,0 +1,419 @@
+<template>
+  <div class="k-tree-list">
+    <draggable
+      v-bind="draggableAttrs"
+      :disabled="disableDrag"
+      :list="internalList"
+      :group="{ name: 'k-tree-list', put: !maxLevelReached }"
+      :move="checkMove"
+      :level="level"
+      @start="onStartDrag"
+      @end="onStopDrag"
+      @change="handleChangeEvent"
+    >
+      <template #item="{ element }">
+        <div
+          :style="indentStyle"
+          :class="{
+            'non-root': level !== 0
+          }"
+          class="k-tree-item-container"
+        >
+          <KTreeItem
+            :key="`tree-item-${key}`"
+            :item="element"
+            :disabled="disableDrag"
+            :class="{
+              'has-children': hasChildren(element)
+            }"
+            @selected="handleSelection"
+          >
+            <template #item-icon>
+              <slot
+                :item="element"
+                name="item-icon"
+              >
+                <KIcon
+                  v-if="element.icon !== 'none'"
+                  :icon="element.icon ? element.icon : 'treeDoc'"
+                  :secondary-color="iconSecondaryColor(element)"
+                  size="24"
+                />
+              </slot>
+            </template>
+            <template #item-label>
+              <slot
+                :item="element"
+                name="item-label"
+              >
+                {{ element.name }}
+              </slot>
+            </template>
+          </KTreeItem>
+          <KTreeList
+            :key="`tree-item-children-${key}`"
+            v-model="element.children"
+            :level="level + 1"
+            :max-level="maxLevel"
+            :disable-drag="disableDrag"
+            :parent-id="element.id"
+            @change="handleChangeEvent"
+            @selected="handleSelection"
+          >
+            <!-- if TS errors appear on these lines switch to `v-slot:` notation and Save to let linter clear -->
+            <template #[itemIcon]="slotProps">
+              <slot
+                v-bind="slotProps"
+                name="item-icon"
+              />
+            </template>
+            <template #[itemLabel]="slotProps">
+              <slot
+                v-bind="slotProps"
+                name="item-label"
+              />
+            </template>
+          </KTreeList>
+        </div>
+      </template>
+    </draggable>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { defineProps, defineEmits, computed, ref, watch, onMounted, PropType } from 'vue'
+import draggable from 'vuedraggable'
```

Comment:
[vue-draggable-next](https://www.npmjs.com/package/vue-draggable-next) has a good nested draggable solution.

Reply:
We are using `vue-draggable@next` is that not the right package? I got confused by this. Should it be `vue-draggable-next` or `vue-draggable@next`?

Reply:
I would opt for `vue-draggable-next`. It's more recently updated and has got better grouping, nesting and transitions

Reply:
I will try doing a straight switch over from `vuedraggable@next` to `vue-draggable-next` but if that doesn't work I'd prefer to stick with `vuedraggable@next` since I've spent a large number of hours getting it to work already.

Reply:
Hooray! We got lucky and the switch over was super easy. We are now using `vue-draggable-next`